src/thunderbots/software/ai/hl/stp/stp.cpp

```diff
@@ -0,0 +1,102 @@
+#include "ai/hl/stp/stp.h"
+
+#include <ai/hl/stp/play/play_factory.h>
+#include <munkres/munkres.h>
+
+#include <chrono>
+#include <exception>
+#include <random>
+
+#include "ai/hl/stp/play/play.h"
+#include "ai/hl/stp/tactic/tactic.h"
+#include "ai/intent/move_intent.h"
+#include "util/logger/init.h"
+
+STP::STP(long random_seed)
+    : random_number_generator(random_seed), uniform_distribution(1, 6)
+{
+}
+
+std::vector<std::unique_ptr<Intent>> STP::getIntentAssignment(const World& world)
+{
+    // Assign a new play if we don't currently have a play assigned, the current play's
+    // invariant no longer holds, or the current play is done
+    if (!current_play || !current_play->invariantHolds(world) || current_play->done())
+    {
+        try
+        {
+            current_play = std::move(calculateNewPlay(world));
```

Comment:
is this just a placeholder? What do we intend to go here?

Reply:
Not sure what you're referring to here? We're calculating the new Play that should be run and assigning it to the `current_play`.

Reply:
Interesting, I'm not sure what I was thinking either.............